# Tarea Clase 4

##  Tarea 1
Une las bases anuales de homicidios: homicidios_2017.csv, homicidios_2018.csv, homicidios_2019.csv, homicidios_2020.csv, homicidios_2021.csv, homicidios_2022.csv, homicidios_2023.csv. Realiza:
- Lee cada archivo usando pd.read_csv().
- A cada base agrega una columna llamada "Año".
- Une todas las bases en un solo DataFrame llamado df_homicidios.
- Crea una columna llamada "Total" (Total = Mujeres + Hombres + No_esp)
- Preguntas para responder:
  - ¿Cuántos registros tiene la base completa?
  - ¿Cuántos homicidios se registraron en total entre 2017 y 2023?
  - ¿Qué año tuvo el mayor número total de homicidios?
  - ¿Qué año tuvo el menor número total de homicidios?
  

In [16]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

ruta2017="https://raw.githubusercontent.com/jamc88/min-curso/refs/heads/main/Datos/homicidios_2017.csv"
ruta2018="https://raw.githubusercontent.com/jamc88/min-curso/refs/heads/main/Datos/homicidios_2018.csv"
ruta2019="https://raw.githubusercontent.com/jamc88/min-curso/refs/heads/main/Datos/homicidios_2019.csv"
ruta2020="https://raw.githubusercontent.com/jamc88/min-curso/refs/heads/main/Datos/homicidios_2020.csv"
ruta2021="https://raw.githubusercontent.com/jamc88/min-curso/refs/heads/main/Datos/homicidios_2021.csv"
ruta2022="https://raw.githubusercontent.com/jamc88/min-curso/refs/heads/main/Datos/homicidios_2022.csv"
ruta2023="https://raw.githubusercontent.com/jamc88/min-curso/refs/heads/main/Datos/homicidios_2023.csv"


In [21]:
archivos = [
(ruta2017, 2017),
(ruta2018, 2018),
(ruta2019, 2019),
(ruta2020, 2020),
(ruta2021, 2021),
(ruta2022, 2022),
(ruta2023, 2023)
]

bases=[]
for archivo,año in archivos:
    df=pd.read_csv(archivo) # Lee cada csv
    df["Año"]=año           # A cada df le agrega la columna "Año"
    bases.append(df)        # Agregar a bases la base actual

#print(bases)  

# Para unir varios DataFrames pd.concat. Para reiniciar índice usar ignore_index=True
df_homicidios=pd.concat(bases,ignore_index=True)

df_homicidios



,Estado,Abreviatura,Regiones,Mujeres,Hombres,No_esp,Año
0,Aguascalientes,AG,Noreste,8,76,0,2017
1,Baja California,BC,Noroeste,205,1955,9,2017
2,Baja California Sur,BS,Noroeste,77,673,1,2017
3,Campeche,CM,Sureste,6,72,0,2017
4,Coahuila de Zaragoza,CO,Noreste,41,234,0,2017
...,...,...,...,...,...,...,...
219,Tamaulipas,TM,Noreste,56,476,28,2023
220,Tlaxcala,TL,Centro-Sur,17,122,0,2023
221,Veracruz,VE,Sureste,89,698,2,2023
222,Yucatan,YU,Sureste,13,39,0,2023


In [22]:
# Crea una columna llamada "Total" (Total = Mujeres + Hombres + No_esp)
df_homicidios["Total"]=df_homicidios["Mujeres"]+df_homicidios["Hombres"]+df_homicidios["No_esp"]
df_homicidios.head(3)

,Estado,Abreviatura,Regiones,Mujeres,Hombres,No_esp,Año,Total
0,Aguascalientes,AG,Noreste,8,76,0,2017,84
1,Baja California,BC,Noroeste,205,1955,9,2017,2169
2,Baja California Sur,BS,Noroeste,77,673,1,2017,751


### ¿Cuántos registros tiene la base completa?
La base completa tiene 223 registros contemplando de 2017 a 2023

### ¿Cuántos homicidios se registraron en total entre 2017 y 2023?


In [24]:
print("El número de homicidios totales entre 2017 y 2023 es: ",df_homicidios["Total"].sum())

El número de homicidios totales entre 2017 y 2023 es:  241476


### ¿Qué año tuvo el mayor número total de homicidios?

El año con mayor número de homicidios fue 2018 con 36685

In [42]:
# Haciendo índices los años
hom=df_homicidios.set_index("Año")

#df2017=hom.loc[2017]  #Tomar las filas que tengan etiqueta año y guardar en un dataframe
#df2017["Total"].sum() # Calcular la suma de la columna "Total"

#hom.loc[2017]["Total"].sum()  # Resumiendo los dos pasos anteriores en uno


for i in range(2017,2024):
    homicidios_año=hom.loc[i]["Total"].sum()
    print("Los homicidios en el año ",i," son: ",homicidios_año)



Los homicidios en el año  2017  son:  32079
Los homicidios en el año  2018  son:  36685
Los homicidios en el año  2019  son:  36661
Los homicidios en el año  2020  son:  36773
Los homicidios en el año  2021  son:  35700
Los homicidios en el año  2022  son:  32223
Los homicidios en el año  2023  son:  31355


### ¿Qué año tuvo el menor número total de homicidios?
El año con mayor número de homicidios fue 2023 con 31355

## Tarea 2
Usando la base unida df_homicidios, realiza lo siguiente:
- Calcula el total de homicidios por año.
- Calcula el total de homicidios por estado entre 2017 y 2023.
- Calcula el total de homicidios por región entre 2017 y 2023.
- Identifica los 5 estados con mayor número acumulado de homicidios.
- Grafica la evolución nacional de homicidios por año.
- Preguntas para responder:
  - ¿La tendencia nacional aumenta, disminuye o se mantiene relativamente estable?
  - ¿Cuáles son los 5 estados con más homicidios acumulados?
  - ¿Qué región concentra el mayor número de homicidios?
  - ¿Qué región concentra el menor número de homicidios?
  - ¿Qué conclusiones generales puedes obtener de la gráfica?